pennylane/optimize/gradient_descent.py

```diff
@@ -60,24 +62,22 @@ def step_and_cost(self, objective_fn, *args, grad_fn=None, **kwargs):
             function output prior to the step.
             If single arg is provided, list [array] is replaced by array.
         """
+        argnums = self.argnums_dispath(args, argnums)
```

Comment:
```suggestion
        argnums = self.argnums_dispatch(args, argnums)
```

plus replacing the other parts in the code where this is called